frontend/app/services/wsaddress-service.server.ts

```diff
@@ -0,0 +1,151 @@
+  unitNumber: z.string().optional(),
+  serviceAreaName: z.string().optional(),
+  serviceAreaType: z.string().optional(),
+  serviceAreaQualifier: z.string().optional(),
+  cityLong: z.string().optional(),
+  cityShort: z.string().optional(),
+  deliveryInformation: z.string().optional(),
+  extraInformation: z.string().optional(),
+  statusCode: z.string().optional(),
+  canadaPostInformation: z.array(z.string()).optional(),
+  message: z.string().optional(),
+  addressType: z.string().optional(),
+  streetNumber: z.string().optional(),
+  streetName: z.string().optional(),
+  streetType: z.string().optional(),
+  serviceType: z.string().optional(),
+  serviceNumber: z.string().optional(),
+  country: z.string().optional(),
+  warnings: z.string().optional(),
+  functionalMessages: z.array(z.object({
+    action: z.string(),
+    message: z.string()
+    })
+  ).optional(),
+});
+
+const ValidateWSAddressResponseSchema = z.object({
+  responseType: z.string().optional(),
+  statusCode: z.string().optional(),
+  functionalMessages: z.array(z.object({
+      action: z.string(),
+      message: z.string()
+    })
+  ).optional(),
+  message: z.string().optional(),
+  warnings: z.string().optional(),
+});
+
+
+export const getWSAddressService = moize.promise(createWSAddressService, { onCacheAdd: () => log.info('Creating new WSAddress service') });
+
+async function createWSAddressService() {
+  const { INTEROP_API_BASE_URI } = getEnv();
+
+  async function parseWSAddress({addressLine, city, province, postalCode, country, language}: ParseWSAddressRequestDTOProps) {
+    const url = `${INTEROP_API_BASE_URI}/address/parse`;
+    const response = await fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(new ParseWSAddressRequestDTO(addressLine, city, province, postalCode, country, language)),
+    });
+
+    if (response.ok) {
+      const parseData = ParseWSAddressResponseSchema.parse(await response.json());
+      if(parseData.statusCode === 'Valid') return new ParseWSAddressResponseDTO(
+        parseData.responseType,
+        parseData.addressLine,
+        parseData.city,
+        parseData.province,
+        parseData.postalCode,
+        parseData.streetNumberSuffix,
+        parseData.streetDirection,
+        parseData.unitType,
+        parseData.unitNumber,
+        parseData.serviceAreaName,
+        parseData.serviceAreaType,
+        parseData.serviceAreaQualifier,
+        parseData.cityLong,
+        parseData.cityShort,
+        parseData.deliveryInformation,
+        parseData.extraInformation,
+        parseData.statusCode,
+        parseData.canadaPostInformation,
+        parseData.message,
+        parseData.addressType,
+        parseData.streetNumber,
+        parseData.streetName,
+        parseData.streetType,
+        parseData.serviceType,
+        parseData.serviceNumber,
+        parseData.country,
+        parseData.warnings,
+        parseData.functionalMessages,
+      );
+    }
```

Comment:
If you remove the class implementation of this DTO, it would make working with this data much easier.

For example, the following code should work fine if the class implementation is removed:

``` typescript
if (response.ok) {
  const parseData = ParseWSAddressResponseSchema.parse(await response.json());
  const parsedAddress: ParseWSAddressResponseDTOProps = { ...parseData };
  if(parseData.statusCode === 'Valid') return parsedAddress;
}
```

Reply:
Moved response DTO to wsaddress-service and modified the return